### DISCUSSION TITLE
Telekom Speedport 920v

### DISCUSSION THREAD

**Discussion by openfnord** [Freetz-NG/freetz-ng#678] [Unerfüllte Wünsche]
Telekom Speedport 920v

please add support for Speedport 920v.
So it should be sufficient to clone the hw option of 7570 vdsl 
and then to go for little differences if there are any. 
920v is a fritzbox 7570 vDSL [1]

So a person could just choose the "original Speedport 920v" just to be sure.

Please support all the alien boxes in menuconfig
https://boxmatrix.info/wiki/Property:Freetz-org

[1] https://de.wikipedia.org/wiki/Speedport
"Eine Besonderheit stellt der Speedport W 920V dar. Da dieser bis auf das Gehäuse nahezu baugleich mit der FRITZ!Box Fon WLAN 7570 vDSL ist, lässt sich per FTP die originale Firmware von AVM aufspielen."

**Comment by pfichtner** [comment]
I used to build and run Freetz/Freetz-NG since 2014 and still update/flash fresh builds on it regularly. So W920V already is supported. 

```
--- Hardware/Firmware ----------------------------------------
    Hardware series (All devices)  --->                       
    Hardware type (7570)  --->                                
    Target region (avme - international)  --->                
    Firmware version (FRITZ!OS 04.00+)  --->                  
[*] Compile image for 'alien' hardware                        
      Alien hardware type (Speedport W920V)  --->             
--- Original components --------------------------------------
```
What exactly is what you are missing?

**Comment by openfnord** [comment]
Hi I found it.

It is hidden, like you write.
Why not adding the alien hw in the root view of the menue like that

----alien hardware -----
speedport 920v
1&1 something
O2 something

so one does not need to research which fritzbox the alien box belongs to.

BTW: 
how to allow "non authorized firmware" on 920v?
The router just allows for reboot but does not want to flash evil 3rd party firmware.

Cheers

luja



btw: you can not fownload the firmware from telekom
But I have put it on archive.org


why not having a tar ball with all firmware images that one can untar in ./dl/fw ?
linux also ships fw blobs for all the shity usb devices out there.
So shipping a tar ball with all firmware provides resilience against disapearing firmware.



On December 18, 2022 2:29:46 PM GMT+01:00, Peter Fichtner ***@***.***> wrote:
>I used to build and run Freetz/Freetz-NG since 2014 and still update/flash fresh builds on it regularly. So W920V already is supported. 
>
>```
>--- Hardware/Firmware ----------------------------------------
>    Hardware series (All devices)  --->                       
>    Hardware type (7570)  --->                                
>    Target region (avme - international)  --->                
>    Firmware version (FRITZ!OS 04.00+)  --->                  
>[*] Compile image for 'alien' hardware                        
>      Alien hardware type (Speedport W920V)  --->             
>--- Original components --------------------------------------
>```
>What exactly is what you are missing? 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/Freetz-NG/freetz-ng/discussions/678#discussioncomment-4441851
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
"They who can give up essential liberty to obtain a little temporary
safety, deserve neither liberty nor safety." Benjamin Franklin (1775)

**Reply by pfichtner** [reply]
> BTW:
how to allow "non authorized firmware" on 920v?
The router just allows for reboot but does not want to flash evil 3rd party firmware.

Exactly the way it works for all other FritzBoxes, see wiki/faq, e.g. https://freetz-ng.github.io/freetz-ng/wiki/10_Beginner/basic_questions.en.html

> why not having a tar ball with all firmware images that one can untar in ./dl/fw ?
linux also ships fw blobs for all the shity usb devices out there.
So shipping a tar ball with all firmware provides resilience against disapearing firmware.

Linux is open source, AVM's firmwares are not. It's forbidden to redistribute AVM's firmware images, see https://freetz-ng.github.io/freetz-ng/wiki/10_Beginner/install.en.html

**Comment by openfnord** [comment]
what about a speed2fritz script.
One can find something at archive.org


also the windows $some macro language rukernel tool disappeared.
is there a linux based script or tool to simply flash speedport e.g. 920v. 
they have a adam? bootloader?

why should one use windows?

thanks in advance.

-- 
"They who can give up essential liberty to obtain a little temporary
safety, deserve neither liberty nor safety." Benjamin Franklin (1775)

**Reply by pfichtner** [reply]
> is there a linux based script or tool to simply flash speedport e.g. 920v.
they have a adam? bootloader?

Did you took a look at the links? https://freetz-ng.github.io/freetz-ng/wiki/10_Beginner/basic_questions.en.html
Second dash mentions [push_firmware](https://github.com/Freetz-NG/freetz-ng/blob/master/tools/push_firmware)

> why should one use windows?

Don't know, I never did. Anyone telling you t use it?

**Comment by fda77** [comment]
s2f is no longer developed, readonly mirror: https://github.com/Freetz-NG/speed-to-fritz
